src/modules/flow/selector/selector.c

```diff
@@ -0,0 +1,688 @@
+        get_buf_size(mdata), SOL_BUFFER_FLAGS_FIXED_CAPACITY |
+        SOL_BUFFER_FLAGS_NO_NUL_BYTE);
+    memset(mdata->text_mem, 0, sizeof(*mdata->text_mem));
+}
+
+static int
+selector_open(struct sol_flow_node *node,
+    void *data,
+    const struct sol_flow_node_options *options)
+{
+    int r;
+    struct selector_data *mdata = data;
+    const struct sol_flow_node_type_selector_options *opts =
+        (const struct sol_flow_node_type_selector_options *)options;
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
+        SOL_FLOW_NODE_TYPE_SELECTOR_OPTIONS_API_VERSION, -EINVAL);
+
+    mdata->circular = opts->circular;
+    mdata->node = node;
+
+    if (opts->columns.val <= 0) {
+        SOL_WRN("Selector columns number must be a positive integer, "
+            "but %d was given.", opts->columns.val);
+        return -EINVAL;
+    } else
+        mdata->columns = opts->columns.val;
+
+    if (opts->rows.val <= 0) {
+        SOL_WRN("Selector rows number must be a positive integer, "
+            "but %d was given.", opts->rows.val);
+        return -EINVAL;
+    } else
+        mdata->rows = opts->rows.val;
+
+    mdata->format = strdup(opts->format);
+    SOL_NULL_CHECK(mdata->format, -ENOMEM);
+
+    mdata->title_tag = strstr(mdata->format, TITLE_TAG);
+    mdata->value_tag = strstr(mdata->format, VALUE_TAG);
+
+    if (!mdata->value_tag) {
+        SOL_WRN("Bad format, no {value} tag: %s", mdata->format);
+        r = -EINVAL;
+        goto err_format;
+    }
+
+    if (mdata->title_tag > mdata->value_tag) {
+        SOL_WRN("Bad format, {title} tag placet after {value} tag: %s",
+            mdata->format);
+        r = -EINVAL;
+        goto err_format;
+    }
+
+    mdata->text_mem = calloc(1, get_buf_size(mdata));
+    if (!mdata->text_mem) {
+        r = -ENOMEM;
+        goto err_format;
+    }
+
+    if (opts->title) {
+        mdata->title = strdup(opts->title);
+        if (!mdata->title) {
+            r = -ENOMEM;
+            goto err_title;
+        }
+    }
+
+    if (opts->selection_marker) {
+        mdata->sel_mark = strdup(opts->selection_marker);
+        if (!mdata->sel_mark) {
+            r = -ENOMEM;
+            goto err_sel_mark;
+        }
+    }
+
+    if (opts->cursor_marker) {
+        mdata->cursor_mark = strdup(opts->cursor_marker);
+        if (!mdata->cursor_mark) {
+            r = -ENOMEM;
+            goto err_cursor_mark;
+        }
+    }
+
+    buffer_re_init(mdata);
+
+    sol_ptr_vector_init(&mdata->items);
+    mdata->enabled = true;
+
+    return 0;
+
+err_cursor_mark:
+    free(mdata->sel_mark);
+
+err_sel_mark:
+    free(mdata->title);
+
+err_title:
+    free(mdata->text_mem);
+
+err_format:
+    free(mdata->format);
+    return r;
+}
+
+static void
+selector_close(struct sol_flow_node *node, void *data)
+{
+    struct selector_data *mdata = data;
+    uint16_t i;
+    char *it;
+
+    sol_buffer_fini(&mdata->text_grid);
+
+    free(mdata->format);
```

Comment:
Didn't you forget to free mdata->sel_mark, mdata->title and mdata->text_mem here?
Have you tried to run your samples with valgriind?